syft/workers/base.py

```diff
@@ -739,29 +741,29 @@ def search(self, *query):
                 if query_item == str(key):
                     match = True
 
-                if tensor.tags is not None:
-                    if query_item in tensor.tags:
+                if obj.tags:
+                    if query_item in obj.tags:
                         match = True
 
-                if tensor.description is not None:
-                    if query_item in tensor.description:
+                if obj.description:
```

Comment:
If the description is 0 or an empty string this will return incorrectly. The previous check was correct

Reply:
Thanks for the careful review! Done.